crawl-ref/source/ng-setup.cc

```diff
@@ -235,6 +237,167 @@ static void _give_ammo(weapon_type weapon, int plus)
+    UNRAND_SNAKEBITE,                   // whip "Snakebite"
+    UNRAND_CAPTAIN,                     // captain's cutlass
+    UNRAND_STORM_BOW,                   // storm bow
+    UNRAND_AUGMENTATION,                // robe of Augmentation
+    UNRAND_THIEF,                       // cloak of the Thief
+    UNRAND_DYROVEPREVA,                 // crown of Dyrovepreva
+    UNRAND_BEAR_SPIRIT,                 // hat of the Bear Spirit
+    UNRAND_BOOTS_ASSASSIN,              // boots of the Assassin
+    UNRAND_LEAR,                        // Lear's hauberk
+    UNRAND_ZHOR,                        // skin of Zhor
+    UNRAND_SALAMANDER,                  // salamander hide armour
+    UNRAND_WAR,                         // gauntlets of War
+    UNRAND_RESISTANCE,                  // shield of Resistance
+    UNRAND_FOLLY,                       // robe of Folly
+    UNRAND_MAXWELL,                     // Maxwell's patent armour
+    UNRAND_DRAGONMASK,                  // mask of the Dragon
+    UNRAND_NIGHT,                       // robe of Night
+    UNRAND_DRAGON_KING,                 // armour of the Dragon King
+    UNRAND_ALCHEMIST,                   // hat of the Alchemist
+    UNRAND_FENCERS,                     // fencer's gloves
+    UNRAND_STARLIGHT,                   // cloak of Starlight
+    UNRAND_GONG,                        // shield of the Gong
+    UNRAND_RCLOUDS,                     // robe of Clouds
+    UNRAND_PONDERING,                   // hat of Pondering
+    UNRAND_DEMON_AXE,                   // obsidian axe
+    UNRAND_LIGHTNING_SCALES,            // lightning scales
+    UNRAND_BLACK_KNIGHT_HORSE,          // Black Knight's horse barding
+    UNRAND_AUTUMN_KATANA,               // autumn katana
+    UNRAND_DEVASTATOR,                  // shillelagh "Devastator"
+    UNRAND_WOE,                         // Axe of Woe
+    UNRAND_MOON_TROLL_LEATHER_ARMOUR,   // moon troll leather armour
+    UNRAND_DARK_MAUL,                   // dark maul
+    UNRAND_HIGH_COUNCIL,                // hat of the High Council
+    UNRAND_ARC_BLADE,                   // arc blade
+    UNRAND_SPELLBINDER,                 // demon whip "Spellbinder"
+    UNRAND_ORDER,                       // lajatang of Order
+    UNRAND_FIRESTARTER,                 // great mace "Firestarter"
+    UNRAND_ORANGE_CRYSTAL_PLATE_ARMOUR, // orange crystal plate armour
+    UNRAND_MAJIN,                       // Majin-Bo
+    UNRAND_GYRE,                        // pair of quick blades "Gyre" and "Gimble"
+    UNRAND_ETHERIC_CAGE,                // Maxwell's etheric cage
+    UNRAND_ETERNAL_TORMENT,             // crown of Eternal Torment
+    UNRAND_VINES,                       // robe of Vines
+    UNRAND_KRYIAS,                      // Kryia's mail coat
+    UNRAND_FROSTBITE,                   // frozen axe "Frostbite"
+    UNRAND_TALOS,                       // armour of Talos
+    UNRAND_WARLOCK_MIRROR,              // warlock's mirror
+};
+
```

Comment:
Ermm.... can't you just use art-enum.h and checking them for the nogen flag rather than listing them all out? This seems entirely unnecessary, and if you want to forbid certain unrands from being generated here, you should add a new flag.